### PR TITLE
feat(orderUpdate): Support returnInfo field

### LINKED ISSUE #7: Add ReturnInfo update actions

### PR DESCRIPTION
**Pull request by junajan**
#### Summary
This PR resolves #7 - it adds a basic support for `returnInfo` field.

#### Description
I added a returnInfo action builder together with tests which should check basic use cases.
I also added a test for a case when no update actions are generated and I did some refactor in several places.

#### Reviewers

- [x] Reviewer 1 has approved the PR
- [x] Reviewer 2 has approved the PR

#### TODO

- Tests
    - [x] Unit
    - [x] Integration
    - [ ] Acceptance
- [x] Documentation

### CODE REVIEW

**Comment by junajan** [comment]
@hisabimbola - Probably it does not belong to this PR but still I was wondering if it is correct that we swallow errors here: https://github.com/commercetools/orders-update/blob/master/src/orders-update.js#L59

If there is any error it will be saved in a summary and then it returns `undefined` into next `.then()` function. We should put there `return Promise.reject(error)` if we want to pass it further.

File: src/build-order-actions.js [hunk 1]
```diff
@@ -83,6 +92,55 @@ const buildOrderMethods = {
...
+    if (item.paymentState !== existingItem.paymentState)
+      actions.push({
+        action: 'setReturnPaymentState',
+        returnItemId: item.id,
+        paymentState: item.paymentState,
+      })
+    return actions
+  },
+
+  returnInfo: (order, existingOrder) => {
```
Comment (hisabimbola) [inline src/build-order-actions.js]: I just looked at this now, noticed that the update action was written here, I was expecting the SDK to be used in this case, returninfo sync action is present in the SDK

maybe we take a look

Comment (junajan) [reply src/build-order-actions.js]: I refactored a code and used SDK orderSync module instead of a custom solution.. please check it now.

File: src/orders-update.js [hunk 2]
```diff
@@ -70,13 +70,16 @@ export default class OrdersUpdate {
...
-        state: bluebird.map(lineItem.state, state =>
-          bluebird.props({
-            ...state,
-            fromState: this.getReference(state.fromState, 'state', 'states'),
-            toState: this.getReference(state.toState, 'state', 'states'),
-          }),
-        ),
+        state: bluebird.map(lineItem.state, state => (
+          // do not resolve states when there is no transition
+          state.fromState && state.toState
```
Comment (hisabimbola) [inline src/orders-update.js:75]: 💯

File: src/orders-update.js [hunk 3]
```diff
@@ -143,6 +146,7 @@ export default class OrdersUpdate {
                 version: existingOrder.version,
                 actions,
               })
+              .then(result => result.body)
```
Comment (hisabimbola) [inline src/orders-update.js:149]: why did you change this?

**Review by hisabimbola — APPROVED** [review]

File: src/orders-update.js (hunk 3 above)
Comment (junajan) [reply src/orders-update.js:149]: If there are no changes on an order, script returns `order` object. But when there are some changes it returns API response envelope instead `{ body: order }` so with this we will return an order every time.

**Review by wizzy25 — APPROVED** [review]
LGTM
